### ISSUE TITLE
Archive: enhance mkv tracks metadata

### ISSUE THREAD

**Issue by hekmon** [ByteDream/crunchyroll-go#26]
Hello,

Just wondering why subtitles have their [title](https://github.com/ByteDream/crunchyroll-go/blob/68aa7e903fee57e1d9ba572a77e7f983de99c791/cmd/crunchyroll-go/cmd/archive.go#L548) set to human readable locale but not the their metadata `language` field ? (which is even more important for players to automatically select the right one for their user)

You do set the `language` field for video and audio. Why not subtitles ?

It seems to me that your [actual locales](https://github.com/ByteDream/crunchyroll-go/blob/68aa7e903fee57e1d9ba572a77e7f983de99c791/crunchyroll.go#L20) would be valid values for mkv (from what I see in mkvtoolnix-gui).

Anyway, thank you for your work this software has been working flawlessly so far !

**Comment by hekmon** [comment]
I guess we could extend this bug report to the `language` metadata values set for audio/video track ([here](https://github.com/ByteDream/crunchyroll-go/blob/68aa7e903fee57e1d9ba572a77e7f983de99c791/cmd/crunchyroll-go/cmd/archive.go#L533-L534) and [here](https://github.com/ByteDream/crunchyroll-go/blob/68aa7e903fee57e1d9ba572a77e7f983de99c791/cmd/crunchyroll-go/cmd/archive.go#L541)).

You set it to the [human readable form](https://github.com/ByteDream/crunchyroll-go/blob/68aa7e903fee57e1d9ba572a77e7f983de99c791/utils/locale.go#L33) but ffmpeg expects for matroska a [particular encoding](https://ffmpeg.org/ffmpeg-all.html#Metadata) (ISO 639-2/B).

So I guess the correct line could look like something like this if you want to keep the human readable form:
```go
// video
metadata = append(metadata, fmt.Sprintf("-metadata:s:v:%d", i), fmt.Sprintf("language=%s", locale)
metadata = append(metadata, fmt.Sprintf("-metadata:s:v:%d", i), fmt.Sprintf("title=%s", utils.LocaleLanguage(locale))
// audio
metadata = append(metadata, fmt.Sprintf("-metadata:s:a:%d", i), fmt.Sprintf("language=%s", locale)
metadata = append(metadata, fmt.Sprintf("-metadata:s:a:%d", i), fmt.Sprintf("title=%s", utils.LocaleLanguage(locale))
// subtitle
metadata = append(metadata, fmt.Sprintf("-metadata:s:s:%d", i), fmt.Sprintf("language=%s", locale)
metadata = append(metadata, fmt.Sprintf("-metadata:s:s:%d", i), fmt.Sprintf("title=%s", utils.LocaleLanguage(locale))
```

If ffmpeg accepts the current form of `locale` [values](https://github.com/ByteDream/crunchyroll-go/blob/68aa7e903fee57e1d9ba572a77e7f983de99c791/crunchyroll.go#L20-L30) (or another helper could produce valid values based on the current ones).

**Comment by hekmon** [comment]
Commit [06a5414](https://github.com/ByteDream/crunchyroll-go/commit/06a541421039a50682af40e4e4975dfa1ec9f6c9) seems to work fine, cheers !
![image](https://user-images.githubusercontent.com/3418467/165387545-bfd42b9a-0b33-4434-8483-da46da938dd4.png)

The last thing a bit annoying with subtitle is that without any particular options, the first track will be marked as default which will not necessarily be the one the user reads. I saw you try to detect the user local, it could be a nice addition to set the track language corresponding to the user local as default ?

In case local is not detected or if detected but no corresponding subtitle track can be found, you can still add the `-default_mode infer_no_subs` flag to ffmpeg in order to [not mark any default subtitle track](https://ffmpeg.org/ffmpeg-formats.html#Options-10).

Thanks again for your work.

**Comment by ByteDream** [comment]
Fixed in newest commit 0c92fc0, but I had to use `-disposition:s:0 0` instead of `-default_mode infer_no_subs` since this didn't work for some reason (and `-disposition` is supported by older ffmpeg versions`).

**Comment by hekmon** [comment]
I can confirm 0c92fc0 works fine on my side, thanks !

**Comment by ByteDream** [comment]
> I saw you try to detect the user local, it could be a nice addition to set the track language corresponding to the user local as default ?

This is already implemented since [v2.0.2](https://github.com/ByteDream/crunchyroll-go/releases), I totally forgot about that lol xd. Just set the environment variable `SORT_SUBTITLES=1` before executing and the subtitles should be sorted.
